ckan_cloud_operator/providers/ckan/instance/manager.py

```diff
@@ -83,7 +83,9 @@ def update(instance_id_or_name, override_spec=None, persist_overrides=False, wai
         pre_update_hook_data = deployment_manager.pre_update_hook(instance_id, instance_type, instance, override_spec,
                                                                   skip_route)
 
-        if config_manager.get('use-cloud-native-storage', secret_name=CONFIG_NAME):
+        use_cloud_storage = instance['spec'].get('ckanStorageBucket', {}).get(get_storage_provider_id()) and config_manager.get('use-cloud-native-storage', secret_name=CONFIG_NAME)
```

Comment:
looks like we need to make a function out of it.

Reply:
Better reuse `bucket_credentials` var. Done